pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -522,12 +590,22 @@ private void handleAssignEvent(String serviceUnit, ServiceUnitStateData data) {
     }
 
     private void handleReleaseEvent(String serviceUnit, ServiceUnitStateData data) {
-        if (isTargetBroker(data.sourceBroker())) {
-            ServiceUnitStateData next = new ServiceUnitStateData(Owned, data.broker(), data.sourceBroker());
-            // TODO: when close, pass message to clients to connect to the new broker
-            closeServiceUnit(serviceUnit)
-                    .thenCompose(__ -> pubAsync(serviceUnit, next))
-                    .whenComplete((__, e) -> log(e, serviceUnit, data, next));
+
+        if (isTransferCommand(data)) {
+            if (isTargetBroker(data.sourceBroker())) {
+                ServiceUnitStateData next = new ServiceUnitStateData(Owned, data.broker(), data.sourceBroker());
+                // TODO: when close, pass message to clients to connect to the new broker
+                closeServiceUnit(serviceUnit)
+                        .thenCompose(__ -> pubAsync(serviceUnit, next))
+                        .whenComplete((__, e) -> log(e, serviceUnit, data, next));
+            }
+        } else {
+            if (isTargetBroker(data.broker())) {
```

Comment:
I have been thinking, should we rename the `sourceBroker` to `destBroker`, since the `broker` will usually be treated as "sourceBroker".

Then we can change the logic like this:

```
        if (isTargetBroker(data.broker())) {
            if (isTransferCommand(data)) {
                ServiceUnitStateData next = new ServiceUnitStateData(Owned, data.broker(), data.destBroker());
                // TODO: when close, pass message to clients to connect to the new broker
                closeServiceUnit(serviceUnit)
                        .thenCompose(__ -> pubAsync(serviceUnit, next))
                        .whenComplete((__, e) -> log(e, serviceUnit, data, next));
            } else {
                ServiceUnitStateData next = new ServiceUnitStateData(Free, data.broker());
                closeServiceUnit(serviceUnit)
                        .thenCompose(__ -> pubAsync(serviceUnit, next))
                        .whenComplete((__, e) -> log(e, serviceUnit, data, next));
            }
        }
```

Reply:
`sourceBroker` and `(dest)broker` are Transfer specific properties.

Currently, `broker` means the target/dest broker. For example, Init->Assign transition, there is no source broker.

I have considered making ServiceUnitStateData generic with command-specific properties, but let's not over-engineer this now.

SeviceUnitType {Bundle, NameSpace}
Map<String, Object> properties